x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1787,4 +1848,32 @@ private static LogicalPlan planWithoutSyntheticAttributes(LogicalPlan plan) {
             return newOutput.size() == output.size() ? plan : new Project(Source.EMPTY, plan, newOutput);
         }
     }
+
+    /**
+     * Cast the union typed fields in EsRelation to date_nanos if they are mixed date and date_nanos types.
+     */
+    private static class ResolveUnionTypesInEsRelation extends Rule<LogicalPlan, LogicalPlan> {
```

Comment:
The name and the rule comment are disjoined, suggest renaming the rule to `DateMillisToNanosInEsRelation`.